src/main/java/org/folio/linked/data/validation/dto/LccnPatternValidator.java

```diff
@@ -21,10 +21,18 @@ public class LccnPatternValidator implements ConstraintValidator<LccnPatternCons
   private static final Pattern LCCN_STRUCTURE_B_PATTERN = Pattern.compile("( {2}|[a-z][ a-z])\\d{10}");
 
   private final SpecProvider specProvider;
+  private final LccnNormalizer<String> lccnNormalizer;
+
+  public LccnPatternValidator(SpecProvider specProvider, List<LccnNormalizer<String>> lccnNormalizers) {
+    this.specProvider = specProvider;
+    this.lccnNormalizer = lccnNormalizers.stream().reduce(LccnNormalizer.identity(), LccnNormalizer::andThen);
```

Comment:
Hi @askhat-abishev,
Eventhough this code will work, I got little confused in the beginning. This gave me an impression that first we will apply StructureA normalizer and then pass the output to StructureB normalizer. This is not correct. We have to apply either one of the normalizer based on the pattern of the incoming LCCN.

So, I think a more clean structure for LccnNormalizer is as follows

```
public interface LccnNormalizer extends Predicate<String>, UnaryOperator<String> {
  // Return normalized LCCN value if it is valid, otherwise return empty Optional
  default Optional<String> normalize(String t) {
    if (this.test(t)) {
      return Optional.of(this.apply(t));
    }
    return Optional.empty();
  }
}
```

Then apply the normalization in `LccnPatternValidator` as follows
```
  private String normalize(String lccn) {
    return lccnNormalizers
      .stream()
      .flatMap(normalizer -> normalizer.normalize(lccn).stream())
      .findFirst()
      .orElse(lccn);
  }
```